Source/buildbindinggo.go

```diff
@@ -724,7 +749,7 @@ func getGoType(paramType, namespace, paramClass, paramName string, isPtr bool) (
 		tp.Type = ptrStr + "string"
 		tp.CType = ptrStr + "*C.char"
 		tp.CToGo = ""
-		tp.GoToC = fmt.Sprintf("(%s)(unsafe.Pointer(&[]byte(%s)[0]))", tp.CType, paramName)
+		tp.GoToC = fmt.Sprintf("C.CString(%s)", paramName)
```

Comment:
`C.CString` is allocated in the C heap using malloc and it should be freed after used, so this change will leak all string casts.
I believe there is no need to allocate a new c string when a c function requires one and it should be safe to just pass the reference to the memory backing the go string, it won't be relocated nor freed at least until the go wrapper function finishes.
When casting a string from c to go we are just taking a temporary

Reply:
The issue with the old code was the the string was not nullterminated, causing a lot of memory issues.

Reply:
Good point. I didn't remember that go string are not null-terminated. There are two option then:

- Append a null termination char to the end of the string: `tp.GoToC = fmt.Sprintf("(%s)(unsafe.Pointer(&[]byte(%s`+"\x00")[0]))", tp.CType, paramName)`. It is unsafe and I don't know if it will always work.
- Use `C.CString(...)` and `defer C.free(...)` in the next line.